packages/transaction-controller/src/utils/swaps.ts

```diff
@@ -338,7 +333,8 @@ function updateSwapApprovalTransaction(
     type,
     sourceTokenSymbol,
   }) as Partial<TransactionMeta>;
-  merge(transactionMeta, swapApprovalTransaction);
```

Comment:
`merge` directly modifies `transactionMeta`, and we can't do that anymore. We don't need that complexity and can get away with a simple spread.